### ISSUE TITLE
Deprecating boost and implementing C++11 support

### ISSUE THREAD

**Issue by GoogleCodeExporter** [jbeder/yaml-cpp#264]
```
Requiring the adoption of the boost libraries so as to use yaml-cpp is 
problematic for many developers.

Given that the C++11 standard library now includes all the features yaml-cpp 
currently uses from boost, it would be good to deprecate the use of boost in 
favour of C++11.

```

Original issue reported on code.google.com by `bruno.j....@gmail.com` on 12 Nov 2014 at 3:03

**Comment by blair1618** [comment]
I've been working on removing Boost dependencies from yaml-cpp: 

https://github.com/blair1618/yaml-cpp/tree/no-boost

I tried to duplicate the functionality of the Boost iterator_adaptor and iterator_facade using the std::iterator interface. The library and tests all compile and most of the tests pass, but a few of the tests cause segfaults so I must be overlooking some (hopefully small) bad memory usage. For anyone who is interested in this feature, a fresh set of eyes will help!

**Comment by blair1618** [comment]
I've fully replaced Boost now, with all tests passing! 

https://github.com/jbeder/yaml-cpp/pull/309

Clang 3.6 and GCC 4.9 support all needed C++11 features, but I've yet to determine compatibility with other toolchains.

**Comment by kamil-kolaczynski** [comment]
@blair1618 
Thank you very much for sharing yaml-cpp without the Boost version!

**Comment by kramasamy** [comment]
Thanks for creating yaml-cpp without the boost. Wondering what is the ETA for this to be in the mainline?

**Comment by jbeder** [comment]
First, @blair1618, thanks for your work here.

Second, @kramasamy, I haven't had time to look at this yet, and I don't generally have a ton of time to work on this library. But next time I get a chunk of time to work on this, I'll look into merging this.

**Comment by kramasamy** [comment]
Thanks @jbeder - eagerly looking forward to it.

**Comment by Dekken** [comment]
@blair1618, there's no issues section for your branch so I'm posting here.
It works with visual studio 13 and 15, one thing I've noticed in my limited testing is that
"a:b" causes a crash where as with gcc it throws an error, should be "a: b" (missing space)

**Comment by kramasamy** [comment]
Another data point - I have but the non boost version of yaml cpp in production. It has been running at a large scale for some time. We did not face any issues so far. Hope to see it in the master soon!

**Comment by blair1618** [comment]
Re-submitted as https://github.com/jbeder/yaml-cpp/pull/342

**Comment by jbeder** [comment]
Fixed 24fa1b33805c9a91df0f32c46c28e314dd7ad96f.
